src/client/particles.cpp

```diff
@@ -81,33 +80,28 @@ Particle::Particle(
 	m_material.setTexture(0, texture);
 	m_texpos = texpos;
 	m_texsize = texsize;
-	m_animation = anim;
+	m_animation = p.animation;
```

Comment:
What would be the disadvantages of keeping `ParticleParameters` inside the class, as replacement for those variables? The diff might be larger, but I think that would be fine.

Reply:
Larger diff was my original reason not to. The code uses and modifies lots of variables that would be inside the structure and I think that doesn't really make sense to keep them together for that.

Reply:
I see. Let's keep it as-is for now.